instrumentation/restlet/restlet-1.0/library/src/main/java/io/opentelemetry/instrumentation/restlet/v1_0/RestletHttpAttributesExtractor.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.instrumentation.restlet.v1_0;
+
+import io.opentelemetry.instrumentation.api.instrumenter.http.HttpAttributesExtractor;
+import io.opentelemetry.semconv.trace.attributes.SemanticAttributes;
+import org.checkerframework.checker.nullness.qual.Nullable;
+import org.restlet.data.Reference;
+import org.restlet.data.Request;
+import org.restlet.data.Response;
+
+final class RestletHttpAttributesExtractor extends HttpAttributesExtractor<Request, Response> {
+  @Override
+  protected @Nullable String method(Request request) {
+    return request.getMethod().toString();
+  }
+
+  @Override
+  protected @Nullable String url(Request request) {
+    return request.getOriginalRef().toString();
+  }
+
+  @Override
+  protected @Nullable String target(Request request) {
+    Reference ref = request.getOriginalRef();
+    String path = ref.getPath();
+    return ref.hasQuery() ? path + "?" + ref.getQuery() : path;
+  }
+
+  @Override
+  protected @Nullable String host(Request request) {
+    return null;
+  }
+
+  @Override
+  protected @Nullable String route(Request request) {
+    return null;
+  }
+
+  @Override
+  protected @Nullable String scheme(Request request) {
+    return request.getOriginalRef().getScheme();
+  }
+
+  @Override
+  protected @Nullable String userAgent(Request request) {
+    return request.getClientInfo().getAgent();
+  }
+
+  @Override
+  protected @Nullable Long requestContentLength(Request request, @Nullable Response response) {
+    return null;
+  }
+
+  @Override
+  protected @Nullable Long requestContentLengthUncompressed(
+      Request request, @Nullable Response response) {
+    return null;
+  }
+
+  @Override
+  protected @Nullable String flavor(Request request, @Nullable Response response) {
+    String version = (String) request.getAttributes().get("org.restlet.http.version");
+    switch (version) {
```

Comment:
Can `version` be null? If so, please add a nullcheck before the `switch` statement.

Reply:
Version is always set in the default implementation of Restlet's `HttpClientCall`